sdk/src/main/java/co/omise/android/Client.java

```diff
@@ -50,8 +60,35 @@ public void run() {
         });
     }
 
+    private X509TrustManager systemDefaultTrustManager() {
+        try {
+            TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(
+                    TrustManagerFactory.getDefaultAlgorithm());
+            trustManagerFactory.init((KeyStore) null);
+            TrustManager[] trustManagers = trustManagerFactory.getTrustManagers();
+            if (trustManagers.length != 1 || !(trustManagers[0] instanceof X509TrustManager)) {
+                throw new IllegalStateException("Unexpected default trust managers:"
+                        + Arrays.toString(trustManagers));
+            }
+            return (X509TrustManager) trustManagers[0];
+        } catch (GeneralSecurityException e) {
+            throw new AssertionError(); // The system has no TLS. Just give up.
+        }
+    }
+
+
     private OkHttpClient buildHttpClient(final String publicKey) {
-        return new OkHttpClient.Builder()
+        OkHttpClient.Builder builder = new OkHttpClient.Builder();
+        X509TrustManager trustManager = systemDefaultTrustManager();
+        if (Build.VERSION.SDK_INT < 21) {
+            try {
+                builder.sslSocketFactory(new TLSSocketFactory(), trustManager);
+            } catch (Exception ex) {
+                ex.printStackTrace();
```

Comment:
Hmm we don't need to even call `systemDefaultTrustmanager()` under `SDK_INT >= 21` right? So let's skip all these shenanigans on platforms that already works correctly.